### ISSUE TITLE
Several G-API sources files use old OpenVINO headers unconditionally 

### ISSUE THREAD

**Issue by ilya-lavrenov** [opencv/opencv#24431]
### System Information

OpenCV version: 4.8.0
Operating System / Platform: Ubuntu 20.04
Compiler & compiler version: GCC 9.3.0

### Detailed description

Following files use old headers `#include <inference_engine.hpp>` unconditionally:
```
opencv/modules/gapi/test/infer/gapi_infer_ie_test.cpp:15
opencv/modules/gapi/src/backends/ie/giebackend.hpp:18
opencv/modules/gapi/src/backends/ie/giebackend/giewrapper.hpp:12
```
Which emits warnings:
```
In file included from opencv/modules/gapi/test/infer/gapi_infer_ie_test.cpp:15:
openvino/src/inference/include/ie/inference_engine.hpp:17:10: warning: #warning ("The Inference Engine API is deprecated and will be removed in the 2024.0 release. For instructions on transitioning to the new API, please refer to https://docs.openvino.ai/latest/openvino_2_0_transition_guide.html") [-Wcpp]
   17 | #        warning("The Inference Engine API is deprecated and will be removed in the 2024.0 release. For instructions on transitioning to the new API, please refer to https://docs.openvino.ai/latest/openvino_2_0_transition_guide.html")
      |          ^~~~~~~
In file included from openvino/src/inference/include/ie/ie_icnn_network.hpp:30,
                 from openvino/src/inference/include/ie/cpp/ie_cnn_network.h:31,
                 from openvino/src/inference/include/ie/cpp/ie_executable_network.hpp:29,
                 from openvino/src/inference/include/ie/ie_core.hpp:28,
                 from openvino/src/inference/include/ie/inference_engine.hpp:22,
                 from opencv/modules/gapi/test/infer/gapi_infer_ie_test.cpp:15:
openvino/src/core/include/ngraph/function.hpp:13:10: warning: #warning ("The nGraph API is deprecated and will be removed in the 2024.0 release. For instructions on transitioning to the new API, please refer to https://docs.openvino.ai/latest/openvino_2_0_transition_guide.html") [-Wcpp]
   13 | #        warning("The nGraph API is deprecated and will be removed in the 2024.0 release. For instructions on transitioning to the new API, please refer to https://docs.openvino.ai/latest/openvino_2_0_transition_guide.html")
      |          ^~~~~~~
```

A recommendation is to use these headers only before 2022.1 version

### Steps to reproduce

Build OpenCV 4.8.0 and with OpenVINO 2023.1.0

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by ilya-lavrenov** [comment]
@dmatveev am I right that you are addressing the issue in your PR https://github.com/opencv/opencv/pull/24379 ?

**Comment by TolyaTalamanov** [comment]
@ilya-lavrenov Hi, we use old headers indeed but only for our `IE` inference backend which is supposed to be functional until the end of API 1.0 era :)

But note that we also have new `OV` backend that is built a top of `OpenVINO API 2.0`.

**Comment by asmorkalov** [comment]
@TolyaTalamanov @ilya-lavrenov What are the next steps here?

**Comment by TolyaTalamanov** [comment]
Hi @asmorkalov, we already hide `API 1.0` under `ifdef` if the OpenVINO >= `2024.0`. See [PR](https://github.com/opencv/opencv/pull/24878)

To summarize:
`IE` backend is available for: `2019.1` < **version** < `2024.0`, starts with `2022.1.0` it will produce warnings
`OV` backend is available for: `2022.1.0` <= **version**
